### PR TITLE
Js2

### PR DESCRIPTION
**Pull request by StianSto**
preliminary review: 
all the required features should have been added. some things like better UX and UI handling is needed. see projects for task yet to be completed before delivery.

### CODE REVIEW

**Comment by StianSto** [comment]
Q: is the way i have added routing a good method, or is it unnecessary complicated?

File: README.md [hunk 1]
```diff
@@ -1,2 +1,3 @@
 # SoMeApp_me-and-my-friends
-"Me &amp; My Friends" is a social media website where the user has more control over customization. set their own theme that appears on their banner when posting. This project is a project made das a school project.  
+
```
Comment (Fermain) [inline README.md:2]: How do I run the project? How do I add to it? Which technologies does it use?

File: profile/edit/index.html [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+    ></script>
+    <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin />
+    <link
+      href="https://fonts.googleapis.com/css2?family=Open+Sans:ital,wght@0,400;0,600;0,700;0,800;1,400;1,600;1,700;1,800&display=swap"
+      rel="stylesheet"
+    />
+    <link rel="stylesheet" href="/dist/css/styles.css" />
+    <link rel="preconnect" href="https://fonts.googleapis.com" />
+    <script src="/src/js/script.js" type="module"></script>
+    <script src="/node_modules/bootstrap/dist/js/bootstrap.bundle.min.js"></script>
```
Comment (Fermain) [inline profile/edit/index.html:20]: I think you should add `defer` to both of these.

File: profile/edit/index.html [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+  </head>
+  <body style="--sidebar-width-open: 40vw">
```
Comment (Fermain) [inline profile/edit/index.html:22]: Why can't you do this in SCSS?

File: src/js/api/auth/loginAuth.mjs [hunk 4]
```diff
@@ -0,0 +1,45 @@
+import errMsg from "../../functions/errMsg.mjs";
+import * as storage from "../../storage/index.mjs";
+
+/**
+ * user log in. needs valid email and password.
+ * @param {text} email valid email
```
Comment (Fermain) [inline src/js/api/auth/loginAuth.mjs:6]:
```suggestion
 * @param {string} email valid email
```

File: src/js/api/auth/loginAuth.mjs [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+ * @param {text} password enter valid password
```
Comment (Fermain) [inline src/js/api/auth/loginAuth.mjs:7]:
```suggestion
 * @param {string} password enter valid password
```

File: src/js/api/auth/loginAuth.mjs [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+ * @param {url to api login endpoint} APIUrl url to api login endpoint
```
Comment (Fermain) [inline src/js/api/auth/loginAuth.mjs:8]:
```suggestion
 * @param {string} APIUrl url to api login endpoint
```

**Review by Fermain — APPROVED** [review]
As you note in your intro, there are remaining UX concerns. The code is readable and does not contain errors or mistakes that I could discern. The app itself is attractive, responsive and delivers an uneven but good UX.

The core requirements are met and some of the stretch goals are also met. With this in mind I suggest focusing on the user journey. Make sure to add a redirect to the homepage if the user is not logged in so that they are bounced to the login page.

File: profile/edit/index.html (hunk 2 above)
Comment (StianSto) [reply profile/edit/index.html:20]: you are right, ill add them promptly :)

**Comment by Fermain** [comment]
> Q: is the way i have added routing a good method, or is it unnecessary complicated?

I like this approach and it's similar to what you will see in React projects

File: profile/edit/index.html (hunk 3 above)
Comment (StianSto) [reply profile/edit/index.html:22]: on my website i am using a side-bar for a friendlist. i am using the same sidebar but replacing the friendlist with an edit profile form on "edit profile". the width is different and unique for this page. this way seemed like the least amount of effort required to get the result i wanted. i could maybe change the variable in js?

File: README.md (hunk 1 above)
Comment (StianSto) [reply README.md:2]: [RESOLVED] readme now contains detailed instructions on how to clone, install and run the app. it also includes other relevant info aobut the project.

**Comment by StianSto** [comment]
## bug fixes and improvements
thank you so much for your feedback on Loom. i have spent some time trying to fix and improve my app in line with your feedback :) if you do have some time to spare someday, do take a look. there is no nned to rush with this of course :) i know you are a busy man, so im happy enough to just get this merged into main

#### bugs fixed: 

- filter not appending results
- post-options not working after re-renders (filter, posting etc...)

#### improvements: 
- added an "edit profile" on profile page
- user gets a simple feedback when updating their profile
- user gets feedback when creating a new post. also allows the user to instantly delete or go to edit post page
- user gets feedback when editing a post. 
- added delete post button in post-options

**Pull request merged**